SpeechSandbox/Assets/Scripts/SpeechSandboxStreaming.cs

```diff
@@ -34,52 +34,140 @@ public class SpeechSandboxStreaming : MonoBehaviour
-    private SpeechToText _speechToText;
-    private Conversation _conversation;
 
+    #region PLEASE SET THESE VARIABLES IN THE INSPECTOR
+    [Header("Speech To Text")]
+    [Tooltip("The service URL (optional). This defaults to \"https://stream.watsonplatform.net/speech-to-text/api\"")]
+    [SerializeField]
+    private string stt_serviceUrl = "";
+    [Header("CF Authentication")]
+    [Tooltip("The authentication username.")]
+    [SerializeField]
     private string stt_username = "";
-    private string stt_password = "";
-    // Change stt_url if different from below
-    private string stt_url = "https://stream.watsonplatform.net/speech-to-text/api";
+    [Tooltip("The authentication password.")]
+    [SerializeField]
+    private string stt_password;
+    [Header("IAM Authentication")]
+    [Tooltip("The IAM apikey.")]
+    [SerializeField]
+    private string stt_iamApikey;
+    [Tooltip("The IAM url used to authenticate the apikey (optional). This defaults to \"https://iam.bluemix.net/identity/token\".")]
+    [SerializeField]
+    private string stt_iamUrl;
+
+    [Header("Watson Assistant")]
+    [Tooltip("The service URL (optional). This defaults to \"https://gateway.watsonplatform.net/conversation/api\"")]
+    [SerializeField]
+    private string asst_serviceUrl;
+    [Tooltip("The workspaceId to run the example.")]
+    [SerializeField]
+    private string asst_workspaceID;
+    [Tooltip("The version date with which you would like to use the service in the form YYYY-MM-DD. Current is 2018-07-10")]
+    [SerializeField]
+    private string asst_versionDate;
+    [Header("CF Authentication")]
+    [Tooltip("The authentication username.")]
+    [SerializeField]
+    private string asst_username;
+    [Tooltip("The authentication password.")]
+    [SerializeField]
+    private string asst_password;
+
+    [Header("IAM Authentication")]
+    [Tooltip("The IAM apikey.")]
+    [SerializeField]
+    private string asst_iamApikey;
+    [Tooltip("The IAM url used to authenticate the apikey (optional). This defaults to \"https://iam.bluemix.net/identity/token\".")]
+    [SerializeField]
+    private string asst_iamUrl;
 
-    private string convo_username = "";
-    private string convo_password = "";
-    // Change convo_url if different from below
-    // Use this for TLS1.0 required by Unity as of 4/25/18
-    private string convo_url = "https://gateway-tls10.watsonplatform.net/conversation/api";
-    // Change  _conversationVersionDate if different from below
-    private string _conversationVersionDate = "2017-05-26";
-    private string convo_workspaceId = "";
+    #endregion
 
-    public Text ResultsField;
 
     private int _recordingRoutine = 0;
     private string _microphoneID = null;
     private AudioClip _recording = null;
     private int _recordingBufferSize = 1;
     private int _recordingHZ = 22050;
 
+    private SpeechToText _speechToText;
+    private Conversation _conversation;
```

Comment:
We should probably refactor this to use the Assistant service for clarity.

Reply:
Yes, but I'll need to do that in a separate patch..looks like quite a few changes.